RUFAS/biophysical/animal/data_types/nutrition_data_structures.py

```diff
@@ -0,0 +1,164 @@
+
+
+@dataclass
+class NutritionSupply:
+    """
+    Energy and nutrition supply for a ration.
+
+    Attributes
+    ----------
+    metabolizable : float
+        Total metabolizable energy in a ration (Mcal).
+    maintenance : float
+        Energy available for maintenance in a ration (Mcal).
+    lactation : float
+        Energy available for lactation in a ration (Mcal).
+    growth : float
+        Energy available for growth in a ration (Mcal).
+    protein : float
+        Metabolizable protein supplied in a ration (g).
+    calcium : float
+        Calcium supplied in a ration (g).
+    phosphorus : float
+        Phosphorus supplied in a ration (g).
+    dry_matter : float
+        Total dry matter content of a ration (kg).
+    ndf_content : float
+        Total neutral detergent fiber (NDF) in the ration (kg).
+    fat_content : float
+        Total fat content in the ration (kg).
+
+    """
+
+    metabolizable: float
+    maintenance: float
+    lactation: float
+    growth: float
+    protein: float
+    calcium: float
+    phosphorus: float
+    dry_matter: float
+    ndf_content: float
+    fat_content: float
+
+
+@dataclass
+class NutritionEvaluationResults:
+    """
+    Results of evaluating whether a ration supplied the required energy and nutrients.
+
+    Attributes
+    ----------
+    total_energy : float | None
+        Surplus or deficit of total energy in a ration (Mcal). Necessary to know for cows, not heifers.
+    maintenance : float
+        Surplus or deficit of energy in a ration for maintenance (Mcal).
+    lactation : float | None
+        Surplus or deficit of lactation in a ration (Mcal). Necessary to know for cows, not heifers.
+    growth : float
+        Surplus or deficit of energy in a ration for growth (Mcal).
+    protein : float
+        Amount of metabolizable protein by which a ration was outside the acceptable bounds (g). If protein was
+        acceptable, this value is 0.0.
+    calcium : float
+        Surplus or deficit of calcium in a ration (g).
+    phosphorus : float
+        Surplus or deficit of phosphorus in a ration (g).
+    dry_matter : float
+        Amount of dry matter by which a ration was outside the acceptable bounds (kg). If dry matter was acceptable,
+        this value is 0.0.
+    ndf : float
+        Surplus or deficit of neutral detergent fiber (NDF) in a ration. If NDF was acceptable, this valus is 0.0.
+    fat : float
+        Surplus or deficit of fat percentage in a ration. If fat percentage was acceptable, this value is 0.0.
+    is_valid_heifer_ration
+    is_valid_cow_ration
+
+    """
+    total_energy: float | None
+    maintenance: float
+    lactation: float | None
+    growth: float
+    protein: float
+    calcium: float
+    phosphorus: float
+    dry_matter: float
+    ndf: float
+    fat: float
+
+    @property
+    def _clamped_values_are_valid(self) -> bool:
```

Comment:
Just to be consistent with the docstrings language. Not sure which supersedes the other.
```suggestion
    def _clamped_values_are_acceptable(self) -> bool:
```